src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -1,278 +1,44 @@
-     */
-    public ReadsDataSource( final Path samFile ) {
-        this(samFile != null ? Arrays.asList(samFile) : null, (SamReaderFactory)null);
-    }
-
-    /**
-     * Initialize this data source with multiple SAM/BAM files and validation stringency SILENT.
-     *
-     * @param samFiles SAM/BAM files, not null.
-     */
-    public ReadsDataSource( final List<Path> samFiles ) {
-        this(samFiles, (SamReaderFactory)null);
-    }
-
-    /**
-     * Initialize this data source with a single SAM/BAM file and a custom SamReaderFactory
-     *
-     * @param samPath path to SAM/BAM file, not null.
-     * @param customSamReaderFactory SamReaderFactory to use, if null a default factory with no reference and validation
-     *                               stringency SILENT is used.
-     */
-    public ReadsDataSource( final Path samPath, SamReaderFactory customSamReaderFactory ) {
-        this(samPath != null ? Arrays.asList(samPath) : null, customSamReaderFactory);
-    }
-
-    /**
-     * Initialize this data source with multiple SAM/BAM files and a custom SamReaderFactory
-     *
-     * @param samPaths path to SAM/BAM file, not null.
-     * @param customSamReaderFactory SamReaderFactory to use, if null a default factory with no reference and validation
-     *                               stringency SILENT is used.
-     */
-    public ReadsDataSource( final List<Path> samPaths, SamReaderFactory customSamReaderFactory ) {
-        this(samPaths, null, customSamReaderFactory, 0, 0);
-    }
-
-    /**
-     * Initialize this data source with multiple SAM/BAM/CRAM files, and explicit indices for those files.
-     *
-     * @param samPaths paths to SAM/BAM/CRAM files, not null
-     * @param samIndices indices for all of the SAM/BAM/CRAM files, in the same order as samPaths. May be null,
-     *                   in which case index paths are inferred automatically.
-     */
-    public ReadsDataSource( final List<Path> samPaths, final List<Path> samIndices ) {
-        this(samPaths, samIndices, null, 0, 0);
-    }
-
-    /**
-     * Initialize this data source with multiple SAM/BAM/CRAM files, explicit indices for those files,
-     * and a custom SamReaderFactory.
-     *
-     * @param samPaths paths to SAM/BAM/CRAM files, not null
-     * @param samIndices indices for all of the SAM/BAM/CRAM files, in the same order as samPaths. May be null,
-     *                   in which case index paths are inferred automatically.
-     * @param customSamReaderFactory SamReaderFactory to use, if null a default factory with no reference and validation
-     *                               stringency SILENT is used.
-     */
-    public ReadsDataSource( final List<Path> samPaths, final List<Path> samIndices,
-        SamReaderFactory customSamReaderFactory) {
-        this(samPaths, samIndices, customSamReaderFactory, 0, 0);
-    }
+public interface ReadsDataSource extends GATKDataSource<GATKRead>, AutoCloseable {
```

Comment:
There are a few additional methods that I think belong in the interface: `traversalIsBounded()`, and `indicesAvailable()` (though we may want to rename the latter to something like `isQueryableByInterval()`)

Reply:
Neither of those are used outside of the class and it's unit tests so I left them out.  We could add an isQueryable.

Reply:
They complement the existing interface methods like `setTraversalBounds()`, `query()` (which throws if `! indicesAvailable()`), which is why I think they belong up in the interface.

Reply:
I pulled up traversalIsBounded and added a new isQueryableByInterval